eng/packages/http-client-csharp-mgmt/generator/TestProjects/Local/Mgmt-TypeSpec/src/Generated/Extensions/MgmtTypeSpecExtensions.cs

```diff
@@ -71,10 +72,11 @@ public static Response<FooResource> GetFoo(this ResourceGroupResource resourceGr
         /// <param name="fooName"> The name of the Foo. </param>
         /// <param name="cancellationToken"> The cancellation token that can be used to cancel the operation. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="resourceGroupResource"/> or <paramref name="fooName"/> is null. </exception>
+        /// <exception cref="ArgumentException"> <paramref name="fooName"/> is an empty string, and was expected to be non-empty. </exception>
         public static async Task<Response<FooResource>> GetFooAsync(this ResourceGroupResource resourceGroupResource, string fooName, CancellationToken cancellationToken = default)
         {
             Argument.AssertNotNull(resourceGroupResource, nameof(resourceGroupResource));
-            Argument.AssertNotNull(fooName, nameof(fooName));
+            Argument.AssertNotNullOrEmpty(fooName, nameof(fooName));
```

Comment:
I think this change is fine, but I wonder where this change comes from?

Reply:
This was a fix in MTG. We previously were not checking for empty for required string parameters. The new behavior matches autorest.csharp behavior.

Reply:
https://github.com/microsoft/typespec/pull/7599